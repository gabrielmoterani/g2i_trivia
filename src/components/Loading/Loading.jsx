// React Imports
import React from 'react';
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';

// Static Imports
import COLORS from '../../config/theme';

const Loading = () => {
    return ( 
        <View style={[styles.container, styles.horizontal]}>
            <ActivityIndicator size="large" color={COLORS.background} />
      </View>
     );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
    },
    horizontal: {
      flexDirection: 'row',
      justifyContent: 'space-around',
      padding: 10,
    },
  });
 
export default Loading;