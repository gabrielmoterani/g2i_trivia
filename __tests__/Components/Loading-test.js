import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../../src/components/Loading';

const setup = () => shallow(<Loading />);


describe('Test LoadingComponent', () => {
    const wrapper = setup();
  
    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render spinner', () => {
        const loading = wrapper.find('ActivityIndicator')
        expect(loading).toExist()
    });

});