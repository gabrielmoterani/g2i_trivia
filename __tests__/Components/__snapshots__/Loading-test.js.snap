// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoadingComponent should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#f00d00"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
